Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs

```diff
@@ -1304,8 +1308,14 @@ internal protected void OnAssemblyLoaded (string assemblyLocation)
 		{
 			AssemblyLoaded?.Invoke (this, new AssemblyEventArgs (assemblyLocation));
 		}
-		
-		internal protected void SetBusyState (BusyStateEventArgs args)
+
+		internal protected void OnAssemblyAdded (Assembly assembly)
+		{
+			assemblies.Add (assembly);
```

Comment:
Assemblies can also be unloaded during a session, so this list needs to support assembly removal as well.